### PR TITLE
IGNITE-20388: Migrate ClusterPerClassIntegrationTest to run queries via public api

### PR DESCRIPTION
**Pull request by ygerzhedovich**
Thank you for submitting the pull request.

To streamline the review process of the patch and ensure better code quality
we ask both an author and a reviewer to verify the following:

### The Review Checklist
- [ ] **Formal criteria:** TC status, codestyle, mandatory documentation. Also make sure to complete the following:  
\- There is a single JIRA ticket related to the pull request.  
\- The web-link to the pull request is attached to the JIRA ticket.  
\- The JIRA ticket has the Patch Available state.  
\- The description of the JIRA ticket explains WHAT was made, WHY and HOW.  
\- The pull request title is treated as the final commit message. The following pattern must be used: IGNITE-XXXX Change summary where XXXX - number of JIRA issue.
- [ ] **Design:** new code conforms with the design principles of the components it is added to.
- [ ] **Patch quality:** patch cannot be split into smaller pieces, its size must be reasonable.
- [ ] **Code quality:** code is clean and readable, necessary developer documentation is added if needed.
- [ ] **Tests code quality:** test set covers positive/negative scenarios, happy/edge cases. Tests are effective in terms of execution time and resources.

### Notes
- [Apache Ignite Coding Guidelines](https://cwiki.apache.org/confluence/display/IGNITE/Java+Code+Style+Guide)

### CODE REVIEW

**Review by AMashenkov — APPROVED** [review]

File: modules/sql-engine/src/testFixtures/java/org/apache/ignite/internal/sql/engine/util/SqlTestUtils.java [hunk 1]
```diff
@@ -200,4 +208,40 @@ public static Object generateValueByType(int base, ColumnType type) {
...
+     * @param args Dynamic parameters for a given query.
+     * @return List of lists , where outer list represents a rows, internal lists represents a columns.
+     */
+    public static List<List<Object>> sql(Ignite ignite, @Nullable Transaction tx, String sql, Object... args) {
+        try (Session session = ignite.sql().createSession()) {
+            ResultSet<SqlRow> rs = session.execute(tx, sql, args);
+
+            return getAllResultSet(rs);
+        }
+    }
```
Comment (korlov42) [inline modules/sql-engine/src/testFixtures/java/org/apache/ignite/internal/sql/engine/util/SqlTestUtils.java:227]:
```suggestion
    public static List<List<Object>> sql(Ignite ignite, @Nullable Transaction tx, String sql, Object... args) {
        try (
                Session session = ignite.sql().createSession();
                ResultSet<SqlRow> rs = session.execute(tx, sql, args)
        ) {
            return getAllResultSet(rs);
        }
    }
```

File: modules/sql-engine/src/testFixtures/java/org/apache/ignite/internal/sql/engine/util/SqlTestUtils.java [hunk 2]
```diff
@@ -200,4 +208,40 @@ public static Object generateValueByType(int base, ColumnType type) {
                 throw new IllegalArgumentException("unsupported type " + type);
         }
     }
+
+    /**
+     * Run SQL in given Ignite instance with given transaction and parameters.
```
Comment (korlov42) [inline modules/sql-engine/src/testFixtures/java/org/apache/ignite/internal/sql/engine/util/SqlTestUtils.java:213]:
```suggestion
     * Run SQL on given Ignite instance with given transaction and parameters.
```

File: modules/sql-engine/src/testFixtures/java/org/apache/ignite/internal/sql/engine/util/SqlTestUtils.java [hunk 3]
```diff
@@ -200,4 +208,40 @@ public static Object generateValueByType(int base, ColumnType type) {
...
+     *
+     * @param ignite Ignite instance to run a query.
+     * @param tx Transaction to run a given query. Can be {@code null} to run within implicit transaction.
+     * @param sql Query to be run.
+     * @param args Dynamic parameters for a given query.
+     * @return List of lists , where outer list represents a rows, internal lists represents a columns.
```
Comment (korlov42) [inline modules/sql-engine/src/testFixtures/java/org/apache/ignite/internal/sql/engine/util/SqlTestUtils.java:219]:
```suggestion
     * @return List of lists, where outer list represents a rows, internal lists represents a columns.
```

File: modules/runner/src/integrationTest/java/org/apache/ignite/internal/sql/engine/datatypes/tests/DataTypeTestSpec.java [hunk 4]
```diff
@@ -104,6 +104,14 @@ public final Class<?> storageType() {
      * */
     public abstract T wrapIfNecessary(Object storageValue);
 
+    /**
+     * Unwraps {@link NativeTypeWrapper comparable wrapper} into a {@link NativeType}.
+     * If passed values in not wrapped then this method should return original instance.
+     * */
```
Comment (korlov42) [inline modules/runner/src/integrationTest/java/org/apache/ignite/internal/sql/engine/datatypes/tests/DataTypeTestSpec.java:110]:
```suggestion
     * Unwraps {@link NativeTypeWrapper comparable wrapper} into a {@link NativeType}.
     *
     * <p>If passed values are not wrapped, then this method should return original instance.
     */
```

**Review by korlov42 — APPROVED** [review]

File: modules/sql-engine/src/testFixtures/java/org/apache/ignite/internal/sql/engine/util/SqlTestUtils.java (hunk 1 above)
Comment (ygerzhedovich) [reply modules/sql-engine/src/testFixtures/java/org/apache/ignite/internal/sql/engine/util/SqlTestUtils.java]: agree, thanks

File: modules/sql-engine/src/testFixtures/java/org/apache/ignite/internal/sql/engine/util/SqlTestUtils.java (hunk 3 above)
Comment (ygerzhedovich) [reply modules/sql-engine/src/testFixtures/java/org/apache/ignite/internal/sql/engine/util/SqlTestUtils.java]: done

**Review by xtern — APPROVED** [review]

**Pull request merged**